src/docfx/build/dependency/DependencyResolver.cs

```diff
@@ -145,7 +146,20 @@ public string GetRelativeUrl(Document fileRelativeTo, Document file)
             // Cannot resolve the file, leave href as is
             if (file is null)
             {
-                return (error, href, fragment, hrefType, null);
+                if (string.IsNullOrEmpty(pathToDocset))
+                {
+                    return (error, href, fragment, hrefType, null);
+                }
+
+                var (errorFromHistory, resourceFromHistory) = TryResolveResourceFromHistory(_gitCommitProvider, relativeTo.Docset, pathToDocset);
+                if (errorFromHistory != null || resourceFromHistory == null)
+                {
+                    return (errorFromHistory ?? error, href, fragment, hrefType, null);
+                }
+
+                // set file to resource got from histroy, reset the error
+                file = resourceFromHistory;
+                error = null;
```

Comment:
why do we need to reset error here?

Reply:
otherwise the previous "file-not-found" error will be returned and reported

Reply:
what about reuse them? `(error, file) = TryResolveResourceFromHistory(_gitCommitProvider, relativeTo.Docset, pathToDocset);`

Reply:
error can't be reused, when file is not resolved from history, error maybe null.